### ISSUE TITLE
Infer inclusion of .env files in Docker run tasks

### ISSUE THREAD

**Issue by philliphoff** [microsoft/vscode-docker#1332]
Currently our scaffolded `.dockerignore` ignores `.env` files (for good reason). However, no equivalent files or environment variables are included or created by default when executing Docker run tasks, which can cause the applications to fail to start.

We should consider automatically including `.env` files in the Docker run command (e.g. if no files are otherwise specified), to better the chances of successfully starting applications in their Docker containers.

**Comment by bwateratmsft** [comment]
@philliphoff Docker doesn't do that automatically though; I feel we'd be setting them up for failure later on, with confusion as to why it works in VSCode but not otherwise. Perhaps a warning in the output window if we detect ENV files and they are not in the `docker-run` task definition?

**Comment by rcjsuen** [comment]
I agree with @bwateratmsft. Be careful how you tread here as you will confuse users over moby/moby#28617.

**Comment by philliphoff** [comment]
The issue came out of testing where a local debug session "just worked" due to the existence of an `.env` file whereas the Docker debug session did not, because the environment variable was not set within the container.  I agree that just including the `.env` within the container may not be the best idea, but the issue was intended to be just a starting point for thought around the best way to manage known/likely environment variables needed by applications.

We could, for example, just pre-scaffold environment variable properties found in a `.env` file in the run task definition, or pre-scaffold the inclusion of the `.env` file as a whole.

**Comment by bwateratmsft** [comment]
I like that idea (pre-scaffold including the `.env` file if it is present). This keeps us from doing magic and also shows the user an extensibility point they might not stumble upon on their own.

**Comment by bwateratmsft** [comment]
Closing this for now; the interest doesn't seem to be there.
